devito/ir/stree/algorithms.py

```diff
@@ -131,8 +129,7 @@ def __init__(self, node):
             self.nodes = [node]
 
         def is_compatible(self, node):
-            return (self.parent == node.parent
-                    and (self.dim == node.dim or node.dim.is_Sub))
+            return self.parent == node.parent and self.dim.root == node.dim.root
```

Comment:
don't they have aways same root if same parent?

Reply:
no, you could have
```
for t
   for a
      ...
   for b
      ...
```
here the ``a`` and ``b`` subtree have the same parent (the ``t`` subtree), but the dimensions themselves, and therefore their root as well, are completely different